### PR TITLE
Add a CI check for JuliaFormatter

### PR DESCRIPTION
**Pull request by odow**
It's about time we turned this on.

The diff is huge, but that is mostly changes in `/test`.

### CODE REVIEW

File: .github/workflows/format_check.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+          using Pkg
+          Pkg.add(PackageSpec(name="JuliaFormatter", version="0.13.2"))
+          using JuliaFormatter
+          format("src", verbose=true)
+          format("test", verbose=true)
+          out = String(read(Cmd(`git diff --name-only`)))
+          if isempty(out)
+              exit(0)
+          end
+          @error "Some files have not been formatted !!!"
```
Comment (blegat) [inline .github/workflows/format_check.yml:29]: You don't get more details than that ?

Comment (odow) [reply .github/workflows/format_check.yml:29]: We can make it print the diff. For now it just prints which files. But more worryingly, formatting broke the tests...

**Review by blegat — APPROVED** [review]
Looks good but can we merge https://github.com/jump-dev/MathOptInterface.jl/pull/1267 first since they have a lot of conflicts and this one is automatically generated?

**Comment by odow** [comment]
Closing in favor of #1289

**Pull request closed without merging**